master/static/migrations/20220228132402_store-profile-pics.up.sql

```diff
@@ -0,0 +1,22 @@
+CREATE TABLE user_profile_images (
+  id SERIAL PRIMARY KEY,
+  user_id INT,
+  file_data BYTEA NOT NULL,
+  CONSTRAINT photo_for_user
+   FOREIGN KEY(user_id)
+      REFERENCES users(id)
+   ON DELETE CASCADE
+);
+
+ALTER TABLE users ADD COLUMN modified_at TIMESTAMP DEFAULT current_timestamp;
+
+CREATE FUNCTION autoupdate_user_modified() RETURNS trigger AS $$
+BEGIN
+  NEW.modified_at := NOW();
+  RETURN NEW;
+END;
+$$ LANGUAGE plpgsql;
+
+CREATE TRIGGER autoupdate_user_modified
+BEFORE UPDATE ON users
```

Comment:
dont you need to update users on update to user_profile_images

Reply:
this trigger might be better removed then, here are two other options
- we could create two triggers to update users.modified_at when there's an insert or update on the pictures table
- I was planning to update users.modified_at in a separate line of SQL, so we could remove this trigger and update the date directly in the future uploader PR

Reply:
you can write a trigger that does this, it's just a bit more complicated.

> I was planning to update users.modified_at in a separate line of SQL, so we could remove this trigger and update the date directly in the future uploader PR

yeah, when we first talked i thought just manually updating in a tx also works, but then it can be forgotten. it's a tad nice if the data model knows how to keep itself correct. but i'm ok with that approach.

Reply:
Updated the trigger to work on insert/update of user_profile_images.

I was also going to add on DELETE, but then the function will be slightly different, so maybe that would be handled differently.